src/psyclone/psyGen.py

```diff
@@ -3708,9 +3704,59 @@ def __init__(self, call, arg_info, access):
             # There are unit-tests where we create Arguments without an
             # associated call.
             if self._call:
+
+                # Keep original list of arguments
+                previous_arguments = \
+                        self._call.root.symbol_table.argument_list
+
+                # Find the tag to use
                 tag = "AlgArgs_" + self._text
-                self._name = self._call.root.symbol_table.symbol_from_tag(
-                    tag, self._orig_name).name
+
+                # Prepare the Argument Interface Access value
+                if access and access.name == "READ":
+                    argument_access = ArgumentInterface.Access.READ
+                elif access and access.name == "WRITE":
+                    argument_access = ArgumentInterface.Access.WRITE
+                else:
+                    # If access is READWRITE, INC, SUM, UNKNOWN or no access
+                    # is given, use a READWRITE argument interface.
+                    argument_access = ArgumentInterface.Access.READWRITE
+
+                # Find the tag or create a new symbol with expected attributes
+                new_argument = self._call.root.symbol_table.symbol_from_tag(
+                    tag, root_name=self._orig_name, symbol_type=DataSymbol,
+                    datatype=self.infere_datatype(),
+                    interface=ArgumentInterface(argument_access))
+                self._name = new_argument.name
+
+                # Unless the argument already exists with another interface
+                # (e.g. globals) they come from the invoke argument list
+                if isinstance(new_argument.interface, ArgumentInterface):
+                    self._call.root.symbol_table.specify_argument_list(
+                        previous_arguments + [new_argument])
+
+    @abc.abstractmethod
+    def psyir_expression(self):
+        '''
+        :returns: the PSyIR expression represented by this Argument.
+        :rtype: :py:class:`psyclone.psyir.nodes.Node`
+
+        :raises NotImplementedError: this is an abstract method.
+
+        '''
+        raise NotImplementedError("Abstract method. Implement in the API.")
```

Comment:
I think you can remove this line (and associated bit in the docstring) as you've marked it as an abstractmethod so it cannot be called.

Reply:
deleted